File: app/src/main/java/com/krystianwsul/checkme/domainmodel/CustomTime.java
package com.krystianwsul.checkme.domainmodel;

import android.support.annotation.NonNull;

import com.krystianwsul.checkme.utils.CustomTimeKey;
import com.krystianwsul.checkme.utils.time.DayOfWeek;
import com.krystianwsul.checkme.utils.time.HourMinute;
import com.krystianwsul.checkme.utils.time.Time;

import java.util.Set;
import java.util.TreeMap;

public interface CustomTime extends Time {
    @NonNull
    String getName();

    @NonNull
    TreeMap<DayOfWeek, HourMinute> getHourMinutes();

    @NonNull
    CustomTimeKey getCustomTimeKey();

    void updateRecordOf(@NonNull Set<String> addedFriends, @NonNull Set<String> removedFriends);
}

File: app/src/main/java/com/krystianwsul/checkme/domainmodel/CustomTime.kt
package com.krystianwsul.checkme.domainmodel

import com.krystianwsul.common.time.DayOfWeek
import com.krystianwsul.common.time.HourMinute
import com.krystianwsul.common.time.Time
import com.krystianwsul.common.utils.CustomTimeKey
import java.util.*

interface CustomTime : Time {

    var name: String

    val hourMinutes: TreeMap<DayOfWeek, HourMinute>

    val customTimeKey: CustomTimeKey<*>

    fun setHourMinute(dayOfWeek: DayOfWeek, hourMinute: HourMinute)
}
